scala/e2e-tests/src/test/scala/uk/gov/nationalarchives/e2etests/IngestTestsRunner.scala

```diff
@@ -0,0 +1,77 @@
+package uk.gov.nationalarchives.e2etests
+
+import cats.effect.std.AtomicCell
+import cats.effect.unsafe.implicits.global
+import cats.effect.{IO, Ref}
+import cats.syntax.all.*
+import org.scalatest.ParallelTestExecution
+import org.scalatest.funspec.AnyFunSpec
+import uk.gov.nationalarchives.e2etests.IngestTestsRunner.*
+import uk.gov.nationalarchives.e2etests.StepDefs.SqsMessage
+
+import java.nio.file.{Files, Path}
+import java.util.UUID
+import scala.concurrent.duration.*
+import scala.jdk.CollectionConverters.*
+
+trait IngestTestsRunner extends AnyFunSpec with ParallelTestExecution {
+
+  def runIngestTests(): Unit =
+    Files.list(Path.of(getClass.getResource("/features").getPath))
+      .toList.asScala
+      .map(Files.readString)
+      .toList
+      .flatMap { eachString =>
+        val eachTest = eachString.split("\n\n")
+        eachTest.toList.map { test =>
+          val eachTestLine = test.split("\n").toList
+          ingest(eachTestLine.head) {
+            eachTestLine.tail.traverse { line =>
+              line.split(" ").toList match
+                case head :: next =>
+                  val testString = next.mkString(" ")
+                  head match
+                    case "Given" => testGiven(testString)
+                    case "When" => testWhen(testString)
+                    case "Then" => testThen(testString)
+                    case invalid => IO.raiseError(new Exception(s"Invalid step $invalid"))
+                case Nil => IO.unit
```

Comment:
I will raise an error as a sanity check but I don't think we can ever actually get here now I'm trimming the string because even passing in empty string will stiill yield a list of size one.